todo_app/app.py

```diff
@@ -1,25 +1,62 @@
-
+import json
+from pickle import NONE
 from flask import Flask, render_template, request, redirect, url_for, session
+from flask_login.utils import login_user, login_required
 from todo_app.data import session_items
 from todo_app.data.trello import Trello
 import requests
 import os
+import logging
+from flask_login import LoginManager, UserMixin, login_required, login_user, current_user
+from todo_app.data.user import User
+from oauthlib.oauth2 import WebApplicationClient
+
 
 def create_app():
     app = Flask(__name__)
     app.config.from_object('todo_app.flask_config.Config')
-        
+    login_manager = LoginManager()
+    login_manager.init_app(app)  
     trello=Trello()
-
+    os.environ['OAUTHLIB_INSECURE_TRANSPORT']= "1"      # This allows us to use a plain HTTP callback
+    client_id = os.getenv('Git_Client_ID')   
+    client= WebApplicationClient(client_id)
+    client_secret = os.getenv("Git_Client_secret")
+    token_url = 'https://github.com/login/oauth/access_token'
+    authorize_url = 'https://github.com/login/oauth/authorize'
+    api_base_url = 'https://api.github.com/user' 
+    
+        
+    #redirect_uri = 'http://localhost:8000/authorize'
+    #app.logger.setLevel(app.config['LOG_LEVEL'])
+    
+    
     @app.route('/')
+    @login_required
     def index():
         
-    #Call trello.py get cards from List
+        writerFlag = user_is_writer(current_user)
         items = trello.getCardsfromList()
-        return render_template('index.html', items=items)
-        # item_view_model = ViewModel(items)
-        # render_template('index.html',view_model=item_view_model)
+        return render_template('index.html', items=items, writeFlag= writerFlag)
+        
+    """ def user_is_writer(id):
+        if id=="78860635":
+         return False
+        else:
+          return False """  
+      
       
+    def user_is_writer(current_user):
+        if app.config['LOGIN_DISABLED']:
+              return False
+    
+        myRole = {'x78860635': 'reader','78860635':'writer'}
+        if (myRole.get(current_user.id) == "writer"):
+                return True
+        return False    
+	
+          
+          
     @app.route('/additem', methods =["POST"])
     def add_item():
         trello.addCardtodoList(request.form.get('title'))
```

Comment:
There's nothing on this function that checks the user is a writer before allowing them to write; that means that anyone with basic knowledge of our system can bypass our authorisation! We should add some protection to all of these functions, for example a basic approach might be:
```suggestion
    @app.route('/additem', methods =["POST"])
    def add_item():
        if not user_is_writer(current_user):
            return 'Access Denied', 401
        trello.addCardtodoList(request.form.get('title'))
```

Reply:
added the code to '/add item'